app/models/wds-on-site-notifications/wds-on-site-notification.js

```diff
@@ -77,9 +78,29 @@ export default EmberObject.extend({
 				return notificationTypes.articleCommentAtMention;
 			case 'article-comment-reply-at-mention-notification':
 				return notificationTypes.articleCommentReplyAtMention;
+			case 'message-wall-post-notification':
+				return notificationTypes.messageWallPost;
+			case 'message-wall-reply-notification':
+				return notificationTypes.messageWallReply;
+			case 'wall-post-removed-notification':
+				return notificationTypes.messageWallPostRemoved;
 		}
 	},
 
+	getMetadata(notification) {
+		let metadata =
+			typeof notification.metadata === 'undefined' || notification.metadata === null ? null : notification.metadata;
+		if (typeof metadata === 'string') {
+			try {
+				metadata = JSON.parse(metadata);
+			} catch (e) {
+				// continue regardless of error
```

Comment:
Maybe it's better to return `undefined`, empty object or object with default values, here in catch block

Reply:
We'd like it to match the behavior in unified-platform https://github.com/Wikia/unified-platform/blob/master/extensions/fandom/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotificationsController.js#L91-L101

Reply:
`null` is intentional to mean: this field was handled and there's nothing in it. If it's `undefined`, this could mean either it was set to `undefined` or the field was never handled by the code, and you won't know which one.

Reply:
I agree this is consistent with UCP. The comment was added to keep the linter from complaining.